File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/mapper/HamburgesaMapper.java
package mapper;

import dto.hamburgesa.HamburgesaDto;
import dto.hamburgesa.ListaHamburgesaDto;
import dto.ingrediente.IngredienteDto;
import models.Hamburgesa;
import models.Ingrediente;

import java.util.List;
import java.util.UUID;

public class HamburgesaMapper {
    public static HamburgesaDto toDto(Hamburgesa hamburgesa){
        return new HamburgesaDto(
                String.valueOf(hamburgesa.getId()),
                hamburgesa.getNombre(),
                hamburgesa.getIngredientes().stream().map(i -> IngredienteMapper.toDto(i)).toList()
        );
    }

    public static Hamburgesa toHamburgesa(HamburgesaDto hamburgesaDto){
        return new Hamburgesa(
                UUID.fromString(hamburgesaDto.getId()),
                hamburgesaDto.getNombre(),
                hamburgesaDto.getIngredientes().stream().map(i -> IngredienteMapper.toIngrediente(i)).toList()
        );
    }

    public static ListaHamburgesaDto toDto(List<Hamburgesa> hamburgesas){
        return new ListaHamburgesaDto(hamburgesas.stream().map(hamburgesa -> toDto(hamburgesa)).toList());
    }

    public static List<Hamburgesa> toHamburgesas(ListaHamburgesaDto hamburgesas){
        return hamburgesas.hamburgesasDto.stream().map(i -> toHamburgesa(i)).toList();
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/mapper/HamburgesaMapper.kt
package mapper

import dto.hamburgesa.HamburgesaDto
import dto.hamburgesa.ListaHamburgesasDto
import model.Hamburgesa
import java.util.*

fun Hamburgesa.toDto(): HamburgesaDto {
    return HamburgesaDto(
        id = this.id.toString().trim(),
        nombre = this.nombre.trim(),
        ingredientes = this.ingredientes.map { it.toDto() }
    )
}

fun HamburgesaDto.toHamburgesa(): Hamburgesa {
    return Hamburgesa(
        id = UUID.fromString(this.id),
        nombre = this.nombre,
        ingredientes = this.ingredientes.map { it.toIngrediente() }.toMutableList()
    )
}

fun List<Hamburgesa>.toDto(): ListaHamburgesasDto{
    return ListaHamburgesasDto(map { it.toDto() })
}

fun ListaHamburgesasDto.toHamburgesa(): List<Hamburgesa>{
    return this.hamburgesasDto.map { it.toHamburgesa() }
}